test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/PatternExpressionMatchTests.cs

```diff
@@ -0,0 +1,185 @@
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void Match_TokenSegmentTfmToken_AddsTfmProperty()
+        {
+            // We have a simple parser for tfm tokens and it basically returns whatever was passed to it
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "tfm", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            //this pattern has tfm as a token segment and is not match only
+            //As a result tfm_raw property will be saved for the content item token matching with tfm
+            PatternDefinition pattern = new PatternDefinition("contentItem/{tfm}");
+            PatternExpression expression = new PatternExpression(pattern);
+            var path = "contentItem/this/is/tfm/raw/file.txt";
+            var expectedResult = "this/is/tfm/raw/file.txt";
+            ContentItem matchResult = expression.Match(path, propertyDefinitions);
+            Assert.NotNull(matchResult);
+            matchResult.Properties.TryGetValue("tfm_raw", out object result);
+            Assert.Equal(expectedResult, result);
+        }
+
+        [Fact]
+        public void Match_TokenSegmentMultipleTokens_AddsProperties()
+        {
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "name", new ContentPropertyDefinition("any", parser: (o, t) => o) },
+                { "version", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            PatternDefinition pattern = new PatternDefinition("contentItem/{name}/{version}");
+            PatternExpression expression = new PatternExpression(pattern);
+            var expectedName = "mypackage";
+            var expectedVersion = "1.0.0";
+            var path = $"contentItem/{expectedName}/{expectedVersion}";
+            var matchResult = expression.Match(path, propertyDefinitions);
+            Assert.NotNull(matchResult);
+            matchResult.Properties.TryGetValue("name", out object nameResult);
+            matchResult.Properties.TryGetValue("version", out object versionResult);
+            Assert.Equal(expectedName, nameResult);
+            Assert.Equal(expectedVersion, versionResult);
+        }
+
+        [Fact]
+        public void Match_TokenSegmentCustomDelimiter_ShouldMatch()
+        {
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "name", new ContentPropertyDefinition("any", parser: (o, t) => o) },
+                { "version", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            PatternDefinition pattern = new PatternDefinition("contentItem/{name}--{version}");
+            PatternExpression expression = new PatternExpression(pattern);
+            var path = "contentItem/mypackage--1.0.0/file.dll";
```

Comment:
I think this might be another example of unrealistic inputs. In `ManagedCodeConventions.cs` we have some patterns where we don't want the pattern to match files that match the pattern, except are actually in subdirectories. For example, `lib/{tfm}/{assembly}`. This not should match `lib/net7.0/en-au/my.resources.dll`, but it should match `lib/net7.0/my.dll`. Other patterns, that end in `{any?}` usually do want to match files in subdirectories.

I'm not sure what this test is trying to test. I mean, the has has "custom delimiter" in the name. THat's not something NuGet itself uses, so I think it's questionable usefulness. But, testing the pattern `contentItem/{name}--{version}` against the file `contentItems/mypackage--1.0.0/file.dll`, I'm not sure what to expect, so I don't know if this is correct, or if both the product and test have a bug.